design/automate_release_design.md

```diff
@@ -0,0 +1,204 @@
+
+The change resembles the following:
+
+```yaml
+stage('Package artifacts') {
+   steps {
+      sh 'ci/jenkins_build'
+        
+      archiveArtifacts artifacts: "helm-artifacts/", fingerprint: false, allowEmptyArchive: true
+	 }
+   when { tag "v*" }
+   steps {
+     sh 'ci/dispatch_version_release_event'
+   } 
+}
+```
+
+The  `ci/jenkins_build` script spins up a container and packages the Helm Chart like so:
+
+```bash
+source bin/build_utils
+helm_version=3.3.0
+
+docker run --rm \
+  -v $PWD/helm/secrets-provider:/root/helm/secrets-provider \
+  -v $PWD/helm-artifacts/:/root/helm-artifacts \
+  --workdir /root/helm-artifacts \
+   alpine/helm:${helm_version} package ../helm/secrets-provider
+```
+
+ The `ci/dispatch_version_release_event` script will only run on release builds. A curl request will be sent passing in the packaged Helm chart, dispatching an `version-release` event. This will trigger a workflow in `cyberark/helm-charts` to run.
+
+```bash
+curl -X POST \
+	-H "Authorization: token OAUTHN-TOKEN" \
+	"Accept: application/vnd.github.everest-preview+json" \
+	"Content-Type: application/javascript" \
+	https://api.github.com/repos/cyberark/helm-charts/dispatches
+	-d- '@/helm-artifacts/secrets-provider-v1.1.0.tgz'
+	-d '{"event_type": "version-release"}'
+```
+
+### Additions to `cyberark/helm-chart` Github repository
+
+In the `cyberark/helm-chart` repository we will need to create a Github workflow that listens for `version-release` events. Once received this will trigger a series of actions in response. The following is an incomplete Github workflow that will be added to the `cyberark/helm-chart` repository:
+
+```pseudocode
+name: Push Helm package workflow
+
+on:
+  repository_dispatch:
+    types: [version-release]
+    
+jobs:
+		push-helm-package:
+      runs-on: ubuntu-latest
+      steps:
+        - name: checkout-branch
+	        uses: peterjgrainger/action-create-branch@v1.0.0
+          run: | 
+          	echo "Checking out branch 'version-release'"
+        		
+        		...
+
+       - name: pull-request-action
+          uses: vsoch/pull-request-action@1.0.6
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            BRANCH_PREFIX: "release-/"
+            PULL_REQUEST_FROM_BRANCH: "release-v110"
+            PULL_REQUEST_BRANCH: master
+            PULL_REQUEST_TITLE: "Push Helm chart package for new version release"
+            PULL_REQUEST_ASSIGNEES: sigalsax     	
+```
+
+### Push `edge` release on successful master build
+
+Another key addition part of the release automation effort is pushing an `edge` release to Dockerhub on every green master build. `edge` releases are intended for frequent pushes of content and used for closely monitoring development. This tag should not be given to our customer's to use in production. 
```

Comment:
when will we consume this image? where will it be used?

Reply:
We will consume the image when we want to test the latest features without pushing a release. It can also be given to SEs for early testing and it can be used by @eladkug to have a reliable image without having to manually build it. In our experience this leads to conflicting images and extra work differentiating which image is the most recent build